Migrating ('HLEN', 'HSTRLEN', 'HSCAN') #1024

c-harish opened the pull request:
Migrates HLEN, HSTRLEN & HSCAN commands 

PR Checklist
- [x]  Migrated the evalXXX function with the latest definition
- [x]  Update or add unit tests for the new implementation.
- [x]  All unit tests pass successfully.
- [x]  Ensure all integration tests pass successfully.
- [x] Ensure integration tests are added for the migrated command on multi-threaded resp server.
- [x] Add http, websocket tests for the commands
- [x] Move Integration tests for the respective commands under the RESP integration tests directory from Async directory
- [x] Please validate that the documentation for the respective commands is up to date. If not then consider adding them.


closes #1024

c-harish commented:
Hi, @AshwinKul28 @lucifercr07

HLEN, HSTRLEN and HSCAN are migrated and unit tests for the same are added. Integration tests for HTTP, RESP, websocket are added.
Added new doc for HSTRLEN and present docs for HLEN, HSCAN looks good to me. 
All unit tests and integration tests passed successfully.

Please review the PR and let me know your suggestions.

Thanks

AshwinKul28 commented on integration_tests/commands/resp/hlen_test.go:
Hello, thanks for all the test cases, can you also move existing test cases as it is to this file
Please refer [here](https://github.com/DiceDB/dice/blob/master/integration_tests/commands/async/hlen_test.go)

AshwinKul28 commented on integration_tests/commands/resp/hscan_test.go:
Thanks a lot for all the extensive test cases. But just to make sure your changes are aligned with the master can you please move existing hscan test cases to the resp directory? Please refer [here](https://github.com/DiceDB/dice/blob/master/integration_tests/commands/async/hscan_test.go)

AshwinKul28 commented on integration_tests/commands/resp/hstrlen_test.go:
Thanks a lot for all the extensive test cases. But just to make sure your changes are aligned with the master can you please move existing hstrlen test cases to the resp directory? Please refer [here](https://github.com/DiceDB/dice/blob/master/integration_tests/commands/async/hstrlen_test.go)

AshwinKul28 reviewed (changes requested):
Thanks @c-harish for this commendable effort. Changes looks great. I have left some comments to make these commands production ready, please have a look. 

Also, please make sure to check-in the documentation for HLEN and HSCAN as well. You can follow sample documentation [here]([here](https://github.com/DiceDB/dice/blob/master/docs/sample_command_docs.md))

c-harish commented:
Thanks for the review, @AshwinKul28.

Have updated the docs of commands as per sample documentation, fixed 0 return values in store_eval, added missed async testcases to resp and dropped the async integration tests of HLEN/HSTRLEN/HSCAN.

Thanks.

AshwinKul28 reviewed (approved):
LGTM, good to merge.
